cmd/kubeadm/app/util/endpoint.go

```diff
@@ -0,0 +1,38 @@
+/*
+Copyright 2017 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package util
+
+import (
+	"fmt"
+	"net"
+)
+
+// GenerateMasterEndpoint generates a properly formatted Master URL.
+func GenerateMasterEndpoint(address string, port int32) string {
+	// Generate the properly formatted API server endpoint based on IP address type.
```

Comment:
nit: could this just be included (elaborated on) into the docstring for the function, on L24?